src/main.rs

```diff
@@ -198,6 +195,22 @@ enum Commands {
     },
 }
 
+#[derive(Subcommand, Debug, Clone)]
+enum ListCommands {
+    /// List the available blobs on the running provider.
+    Blobs {
+        /// RPC port of the provider
+        #[clap(long, default_value_t = DEFAULT_RPC_PORT)]
+        rpc_port: u16,
+    },
+    /// List the available collections on the running provider.
+    Collections {
+        /// RPC port of the provider
+        #[clap(long, default_value_t = DEFAULT_RPC_PORT)]
+        rpc_port: u16,
+    },
+}
+
```

Comment:
I'm not a fan of the added depth to the `list` command for two reasons:
1. Don't make the user think about what they want to list. `iroh list` without arguments should list stuff.
2. This breaks the convention established with `iroh get`. Iroh get assumes collections, requires a `--single` flag to fetch a single thing. 

I don't think the `--single` flag is great, but I do think that defaulting to collections is the right choice. a collection with a single item in it is still useful.

**Counter proposal:**
the default for `iroh list` follows the convention of `iroh get`:
* lists collections by default
* a `--blobs` flag lists blobs instead of collections

Reply:
thinking about this some more, I would rather get to a more explicit namespacing, potentially like this

```
iroh collection
  list
  get 
iroh blob
  list 
  get
```

The current `--single` approach is quite odd & confusing imo, and I would really like to avoid extending this further into the api.